internal/common/ingest/ingestion_pipeline.go

```diff
@@ -117,12 +119,19 @@ func (i *IngestionPipeline[T, U]) Run(ctx *armadacontext.Context) error {
 	wg.Add(1)
 
 	if i.consumer == nil {
-		consumer, closePulsar, err := i.subscribe()
+		client, consumer, closePulsar, err := i.subscribe()
 		if err != nil {
 			return err
 		}
 		i.consumer = consumer
 		defer closePulsar()
+
+		if i.pulsarConfig.DelayMonitor != nil && i.pulsarConfig.DelayMonitor.Enabled {
```

Comment:
can we make DelayMonitor  a struct rather than a pointer.  that way we just need to check `pulsarConfig.DelayMonitor.Enabled`